tests/test_pyrflx.py

```diff
@@ -592,16 +630,502 @@ def test_ethernet_parse_binary(self) -> None:
+            b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0a"
+        )
+
+        self.frame.set("Destination", int("FFFFFFFFFFFF", 16))
+        self.frame.set("Source", int("0", 16))
+        self.frame.set("Type_Length_TPID", int("8100", 16))
+        self.frame.set("TPID", int("8100", 16))
+        self.frame.set("TCI", 1)
+        self.frame.set("Type_Length", int("0800", 16))
+        self.frame.set("Payload", payload)
+
+        self.assertTrue(self.frame.valid_message)
+
+        with open("tests/ethernet_vlan_tag.raw", "rb") as file:
+            msg_as_bytes: bytes = file.read()
+
+        print(self.frame.to_bytes.hex())
+        print(msg_as_bytes.hex())
+
+        self.assertEqual(self.frame.to_bytes, msg_as_bytes)
+
+    def test_generating_ethernet_2_vlan_dynamic(self) -> None:
+        pass
+
+    # rflx-in_ethernet-test
+
+    def test_parsing_ipv4_in_ethernet(self) -> None:
+        pass
+
+    def test_generating_ipv4_in_ethernet(self) -> None:
+        pass
+
+    # reflx-in_ipv4_test
+
+    def test_parsing_udp_in_ipv4(self) -> None:
+        pass
+
+    def test_parsing_udp_in_ipv4_in_ethernet(self) -> None:
+        pass
+
+    def test_generating_udp_in_ipv4_in_ethernet(self) -> None:
+
+        with open("tests/ethernet_ipv4_udp.raw", "rb") as file:
+            msg_as_bytes: bytes = file.read()
+
+        self.frame.assign(msg_as_bytes)
+
+        parsed_frame = self.frame.to_bytes
+
+        b = b""
+        for _ in itertools.repeat(None, 18):
+            b += b"\x00"
+
+        self.udp.set("Source_Port", 53)
+        self.udp.set("Destination_Port", 53)
+        self.udp.set("Length", 26)
+        self.udp.set("Checksum", int("014E", 16))
+        self.udp.set("Payload", b)
+        udp_binary = self.udp.to_bytes
+
+        self.ipv4.set("Version", 4)
+        self.ipv4.set("IHL", 5)
+        self.ipv4.set("DSCP", 0)
+        self.ipv4.set("ECN", 0)
+        self.ipv4.set("Total_Length", 46)
+        self.ipv4.set("Identification", 1)
+        self.ipv4.set("Flag_R", "False")
+        self.ipv4.set("Flag_DF", "False")
+        self.ipv4.set("Flag_MF", "False")
+        self.ipv4.set("Fragment_Offset", 0)
+        self.ipv4.set("TTL", 64)
+        self.ipv4.set("Protocol", "PROTOCOL_UDP")
+        self.ipv4.set("Header_Checksum", int("7CBC", 16))
+        self.ipv4.set("Source", int("7f000001", 16))
+        self.ipv4.set("Destination", int("7f000001", 16))
+        self.ipv4.set("Payload", udp_binary)
+        ip_binary = self.ipv4.to_bytes
+
+        self.frame.set("Destination", int("FFFFFFFFFFFF", 16))
+        self.frame.set("Source", int("0", 16))
+        self.frame.set("Type_Length_TPID", int("0800", 16))
+        self.frame.set("Type_Length", int("0800", 16))
+        self.frame.set("Payload", ip_binary)
+
+        self.assertTrue(self.frame.valid_message)
+        self.assertEqual(parsed_frame, self.frame.to_bytes)
+
+    # rflx-in_tlv-tests
+
+    def test_null_in_tlv(self) -> None:
+        pass
```

Comment:
Same as above.

Reply:
Fixed in https://github.com/Componolit/RecordFlux/pull/198/commits/1ab7b6311e8bc2f82f25c3c4b96d70a135337223.